airflow-core/src/airflow/serialization/serialized_objects.py

```diff
@@ -1715,78 +1715,86 @@ def serialize_dag(cls, dag: DAG) -> dict:
+                elif k == "params":
+                    v = cls._deserialize_params_dict(v)
+                elif k == "tags":
+                    v = set(v)
+                # else use v as it is
+
+                object.__setattr__(dag, k, v)
+
+            # Set _task_group
+            if "task_group" in encoded_dag:
+                tg = TaskGroupSerialization.deserialize_task_group(
+                    encoded_dag["task_group"],
+                    None,
+                    dag.task_dict,
+                    dag,
+                )
+                object.__setattr__(dag, "task_group", tg)
+            else:
+                # This must be old data that had no task_group. Create a root TaskGroup and add
+                # all tasks to it.
+                object.__setattr__(dag, "task_group", TaskGroup.create_root(dag))
+                for task in dag.tasks:
+                    dag.task_group.add(task)
+
+            # Set has_on_*_callbacks to True if they exist in Serialized blob as False is the default
+            if "has_on_success_callback" in encoded_dag:
+                dag.has_on_success_callback = True
+            if "has_on_failure_callback" in encoded_dag:
+                dag.has_on_failure_callback = True
+
+            keys_to_set_none = (
+                dag.get_serialized_fields() - encoded_dag.keys() - cls._CONSTRUCTOR_PARAMS.keys()
+            )
+            for k in keys_to_set_none:
+                setattr(dag, k, None)
 
-            object.__setattr__(dag, k, v)
+            for task in dag.task_dict.values():
+                SerializedBaseOperator.set_task_dag_references(task, dag)
 
-        # Set _task_group
-        if "task_group" in encoded_dag:
-            tg = TaskGroupSerialization.deserialize_task_group(
-                encoded_dag["task_group"],
-                None,
-                dag.task_dict,
-                dag,
-            )
-            object.__setattr__(dag, "task_group", tg)
-        else:
-            # This must be old data that had no task_group. Create a root TaskGroup and add
-            # all tasks to it.
-            object.__setattr__(dag, "task_group", TaskGroup.create_root(dag))
-            for task in dag.tasks:
-                dag.task_group.add(task)
-
-        # Set has_on_*_callbacks to True if they exist in Serialized blob as False is the default
-        if "has_on_success_callback" in encoded_dag:
-            dag.has_on_success_callback = True
-        if "has_on_failure_callback" in encoded_dag:
-            dag.has_on_failure_callback = True
-
-        keys_to_set_none = dag.get_serialized_fields() - encoded_dag.keys() - cls._CONSTRUCTOR_PARAMS.keys()
-        for k in keys_to_set_none:
-            setattr(dag, k, None)
-
-        for task in dag.task_dict.values():
-            SerializedBaseOperator.set_task_dag_references(task, dag)
-
-        return dag
+            return dag
+        except (RuntimeError, ValueError, KeyError) as err:
```

Comment:
Does RuntimeError here only intend to catch the case we raise ourselves? I’d rather avoid it if possible; it’s used too often for built-in errors that we may not want to mask.

Reply:
I don't know, basically any unexpected error there is a deserialization error, and we prefer to wrap that into a DeserializationError (with the original stack trace), to show that to the users of the API, instead of having a plain `ValueError/KeyError` with a 500 (Default ValueError and KeyError cannot be handled globally by the api server exception handler).

Raising anything else not handled by the server will end up in `500` and the error buried in the stack trace.